src/transforms/reduce/mod.rs

```diff
@@ -40,6 +40,7 @@ pub struct ReduceConfig {
     /// reduce.
     pub ends_when: Option<AnyCondition>,
     pub starts_when: Option<AnyCondition>,
+    pub flushes_when: Option<AnyCondition>,
```

Comment:
I feel like this name conflicts with the `flushes_after` parameter, but not sure of what a better name would be.